### ISSUE TITLE
Intellisense for never before imported libraries doesnt work

### ISSUE THREAD

**Issue by EldinHb** [sveltejs/language-tools#2193]
### Describe the bug

When you install a Svelte library and then try to use a Svelte component in any .svelte file the intellisense cant find the library. When you add `import {} from 'library';` in any other file intellisense is able to find any component of said library.

Screenshots with https://tabler.io/docs/icons/svelte as the library. Could also reproduce with https://www.skeleton.dev/.

No intellisense for any icon.
<img width="940" alt="image" src="https://github.com/sveltejs/language-tools/assets/8807015/ccc1d7f8-71fc-4f3c-a688-53f97d8f120a">

Intellisense works with `import {} from '@tabler/icons-svelte';` in same file.
<img width="832" alt="image" src="https://github.com/sveltejs/language-tools/assets/8807015/4a0eabfc-ba3e-431a-bfd2-a2207aa5360a">

Intellisense works with `import {} from '@tabler/icons-svelte';` in another unused .svelte file.
<img width="929" alt="image" src="https://github.com/sveltejs/language-tools/assets/8807015/0ae33475-fedd-40d1-a8ed-f8821999bb3e">

### Reproduction

Create new sveltekit project. 

Pick empty project with typescript and no other libraries.

Install https://tabler.io/docs/icons/svelte

See the screenshots in the above description.

### Expected behaviour

Intellisense to work with never before imported libraries.

### System Info

- OS: macOS
- IDE: vscode

### Which package is the issue about?

Svelte for VS Code extension

**Comment by jasonlyu123** [comment]
This is expected because the type definition is only loaded if there is an import to it. Loading arbitrary d.ts files from node_modules might cause some performance problems. In TypeScript projects, the auto-import also behaves like this. 

But in the js projects, the automatic [type acquisition](https://www.typescriptlang.org/tsconfig#typeAcquisition) will also discover some of the direct dependencies in a child process. This is a feature that we didn't support, so I am marking this issue as a feature request instead of a bug.

**Comment by EldinHb** [comment]
Do you mean in regular typescript projects that aren't svelte projects? I've never seen this in regular typescript projects or react projects. It works smoothly and I haven't seen performance issues in those projects.

**Comment by jasonlyu123** [comment]
Yes. It's the same. I tried in a ts file in a plain typescript project with only a tsconfig and in a SvelteKit project. They both don't provide auto-import for newly installed packages. 

js projects, i.e. those without tsconfig.json or those with a jsconfig.json. Those are provided by automatic type acquisition running in another [process](https://en.wikipedia.org/wiki/Process_(computing)). And we don't support that.

**Comment by jasonlyu123** [comment]
Hmm. That's weird. That's different from what I saw in both Windows and Ubuntu WSL. Maybe you have some extension or config that changes the default behaviour?

**Comment by jasonlyu123** [comment]
Ah. Found it. It's https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#smarter-auto-imports.  From the implementation notes. `devDependencies` is excluded from this discovery. That's why I kept not getting it. This is indeed not provided by automatic type acquisition. But unfortunately, this is a TypeScript internal API, and it's not something we can switch on easily. It's still a feature request that needs quite some work to enable.
